### PR TITLE
[SPARK-40784][SQL][TESTS] Check error classes in DDLParserSuite

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
This PR aims to replace 'intercept' with 'Check error classes' in DDLParserSuite. include two files:
A.org.apache.spark.sql.execution.command.DDLParserSuite
B.org.apache.spark.sql.catalyst.parser.DDLParserSuite

### Why are the changes needed?
The changes improve the error framework.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
By running the modified test suite:
```
$ build/sbt "test:testOnly *DDLParserSuite"
```

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @MaxGekk

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala [hunk 1]
```diff
@@ -34,6 +34,10 @@ import org.apache.spark.unsafe.types.{CalendarInterval, UTF8String}
 class DDLParserSuite extends AnalysisTest {
   import CatalystSqlParser._
 
+  private def parseException(sqlText: String): SparkThrowable = {
+    intercept[ParseException](parsePlan(sqlText))
+  }
```
Comment (amaliujia) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala:39]: Is this function can be shared with other Suites (e.g. https://github.com/apache/spark/pull/38265/) so move this one to `AnalysisTest`?

**Comment by MaxGekk** [comment]
@panbingkun Please, don't forget about another `DDLParserSuite` at:
org.apache.spark.sql.execution.command.DDLParserSuite

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala (hunk 1 above)
Comment (panbingkun) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala:39]: Yes, i will move it

**Comment by panbingkun** [comment]
> @panbingkun Please, don't forget about another `DDLParserSuite` at: org.apache.spark.sql.execution.command.DDLParserSuite

ok.

**Comment by panbingkun** [comment]
When I work on it, I found typo: https://github.com/apache/spark/pull/38269
cc @MaxGekk

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala (hunk 1 above)
Comment (panbingkun) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala]: Done

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisTest.scala [hunk 2]
```diff
@@ -204,12 +204,17 @@ trait AnalysisTest extends PlanTest {
...
     messages.foreach { message =>
       assert(e.message.contains(message))
     }
     if (errorClass.isDefined) {
       assert(e.getErrorClass == errorClass.get)
     }
   }
+
+  protected def parseException(parser: String => Any)(
+    sqlText: String): ParseException = {
```
Comment (MaxGekk) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisTest.scala:217]: Should be 4 spaces, see https://github.com/databricks/scala-style-guide#spacing-and-indentation

**Review by MaxGekk — APPROVED** [review]

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisTest.scala (hunk 2 above)
Comment (MaxGekk) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisTest.scala:217]: Let's fix this next time.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Comment by amaliujia** [comment]
LGTM!

**Pull request closed without merging**